libs/transactions/src/lib/services/transactions.http.service.ts

```diff
@@ -1,16 +1,37 @@
 import { Injectable } from '@angular/core';
-import { Transaction } from '../model/transaction';
-import { of } from 'rxjs';
-import { HttpClient, HttpParams } from '@angular/common/http';
+import { GetTransactionsWithPostRequestParams, TransactionClientHttpService, TransactionItem } from '@backbase/data-ang/transactions';
+import { combineLatest, Observable, of } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { ArrangementsService } from './arrangements.service';
 import { TransactionsJourneyConfiguration } from './transactions-journey-config.service';
-import { map, switchMap } from 'rxjs/operators';
 
 @Injectable()
 export class TransactionsHttpService {
-  public transactions$ = of(this.configurationService.pageSize).pipe(
-    map((pageSize) => new HttpParams().append('page', String(0)).append('pageSize', String(pageSize))),
-    switchMap((params) => this.http.get<Transaction[]>('/api/transactions', { params })),
+  public transactions$: Observable<TransactionItem[] | undefined> = combineLatest([
+    this.arrangementsService.arrangementIds$,
+    of(this.configurationService.pageSize),
+  ]).pipe(
+    switchMap(
+      ([arrangements, pageSize]) =>
+      this.transactionsHttpService.getTransactionsWithPost(
+        {
+          transactionListRequest: {
+            arrangementsIds: arrangements,
+            size: pageSize,
```

Comment:
And why can't we have it like this because one of the arguments for request is stream already?
As for me, this way looks more consistent, that outer data dependency for a data.